Description: backport of 6b1044cc3f79736b9d707d7f9f3cc8c3eae65193 to fix weird
text spacing.

--- a/src/gui/text/qtextengine.cpp
+++ b/src/gui/text/qtextengine.cpp
@@ -1807,8 +1807,10 @@
 QT_WARNING_DISABLE_DEPRECATED
         if (!actualFontEngine->supportsSubPixelPositions() || (actualFontEngine->fontDef.styleStrategy & QFont::ForceIntegerMetrics)) {
 QT_WARNING_POP
-            for (uint i = 0; i < num_glyphs; ++i)
+            for (uint i = 0; i < num_glyphs; ++i) {
                 g.advances[i] = g.advances[i].round();
+                g.offsets[i].x = g.offsets[i].x.round();
+            }
         }

         glyphs_shaped += num_glyphs;
